README.md

```diff
@@ -114,3 +114,48 @@ The interesting fields for this use case are:
   the SSO user ID to be sent to Segment
 - `status.compliantUsername` - Contains the username used in the cluster audit
   log.
+
+## Details about the Event Sender
+
+Segment has a [built-in mechanism for removing duplicate events][ES1]. This
+mean that we can safely resend the same event multiple times to increase the
+sending process reliability. The duplicate remove mechanism is based on the
+`messageid` [common message field][ES2]. We can use the `auditID` field of the
+cluster audit log record as the value for this field.
+
+Segment also has a [*batch* call][ES3] that allows for sending multiple events
+within a single API call. There is a limit of 500KB data size per-call, while
+individual event JSON records should not exceed 32KB.
+
+The Splunk [streamstats][ES4] command allows for calculating cumulative sums
+and statistics for the records within a Splunk search query, while its
+`reset_after` options allows for restarting the cumulative calculations under
+given conditions. The [transaction][ES5] command allows for the grouping of
+multiple search records into a single record with multi-value fields while the
+`mv_to_json_array` [eval][ES6] function lets one convert such fields into JSON
+lists. Along with the [json_object][ES7] function these could be used to
+convert Splunk query results into a series of JSON objects where each does not
+exceed a given size.
+
+The architecture for the event sending logic would then be:
+
+1. Repeat the following logic on an hourly basis:
```

Comment:
You might want to remove the `1.` here. Looks unnecessary.

Reply:
done.